reconcile/terraform_tgw_attachments.py

```diff
@@ -0,0 +1,166 @@
+import logging
+import sys
+import json
+
+
+import reconcile.queries as queries
+
+from reconcile.utils.aws_api import AWSApi
+from reconcile.utils.defer import defer
+from reconcile.utils.ocm import OCMMap
+from reconcile.utils.terraform_client import TerraformClient as Terraform
+from reconcile.utils.terrascript_client import TerrascriptClient as Terrascript
+from reconcile.utils.semver_helper import make_semver
+
+
+QONTRACT_INTEGRATION = 'terraform_tgw_attachments'
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+
+
+def build_desired_state_tgw_attachments(clusters, ocm_map, settings):
+    """
+    Fetch state for TGW attachments between a cluster and all TGWs
+    in an account in the same region as the cluster
+    """
+    desired_state = []
+    error = False
+
+    for cluster_info in clusters:
+        cluster = cluster_info['name']
+        ocm = ocm_map.get(cluster)
+        peering_info = cluster_info['peering']
+        peer_connections = peering_info['connections']
+        for peer_connection in peer_connections:
+            # We only care about account-tgw peering providers
+            peer_connection_provider = peer_connection['provider']
+            if not peer_connection_provider == 'account-tgw':
+                continue
+            # accepter is the cluster's AWS account
+            cluster_region = cluster_info['spec']['region']
+            accepter = {
+                'cidr_block': cluster_info['network']['vpc'],
+                'region': cluster_region
+            }
+
+            account = peer_connection['account']
+            # assume_role is the role to assume to provision the
+            # peering connection request, through the accepter AWS account.
+            account['assume_role'] = \
+                ocm.get_aws_infrastructure_access_terraform_assume_role(
+                    cluster,
+                    account['uid'],
+                    account['terraformUsername']
+                )
+            account['assume_region'] = accepter['region']
+            account['assume_cidr'] = accepter['cidr_block']
+            aws_api = AWSApi(1, [account], settings=settings)
+            accepter_vpc_id, accepter_route_table_ids = \
+                aws_api.get_cluster_vpc_id(
+                    account,
+                    route_tables=peer_connection.get('manageRoutes')
+                )
+
+            if accepter_vpc_id is None:
+                logging.error(f'[{cluster} could not find VPC ID for cluster')
+                error = True
+                continue
+            accepter['vpc_id'] = accepter_vpc_id
+            accepter['route_table_ids'] = accepter_route_table_ids
+            accepter['account'] = account
+
+            account_tgws = \
+                aws_api.get_tgws_details(
```

Comment:
this can be empty if it's the first time we're provisioning, right?

Reply:
we are not responsible for provisioning the TGW. we are just discovering them. this list can come back empty

Reply:
no, it cannot... 😂